code/datums/elements/rust.dm

```diff
@@ -0,0 +1,75 @@
+/**
+ * Adding this element to an atom will have it automatically render an overlay.
+ * The overlay can be specified in new as the first paramter; if not set it defaults to rust_overlay's rust_default
+ */
+/datum/element/rust
+	element_flags = ELEMENT_BESPOKE|ELEMENT_DETACH
+	id_arg_index = 2
+	/// The rust image itself, since the icon and icon state are only used as an argument
+	var/image/rust_overlay
+
+/datum/element/rust/Attach(atom/target, _rust_icon = 'icons/effects/rust_overlay.dmi', _rust_icon_state = "rust_default")
+	. = ..()
+	if(!isatom(target))
+		return COMPONENT_INCOMPATIBLE
+	if(!rust_overlay)
+		rust_overlay = image(_rust_icon, _rust_icon_state)
+	ADD_TRAIT(target, TRAIT_RUSTY, src)
+	RegisterSignal(target, COMSIG_ATOM_UPDATE_OVERLAYS, .proc/apply_rust_overlay)
+	RegisterSignal(target, COMSIG_PARENT_EXAMINE, .proc/handle_examine)
+	RegisterSignal(target, list(COMSIG_ATOM_SECONDARY_TOOL_ACT(TOOL_WELDER), COMSIG_ATOM_SECONDARY_TOOL_ACT(TOOL_RUSTSCRAPER)), .proc/secondary_tool_act)
+	// Unfortunately registering with parent sometimes doesn't cause an overlay update
+	target.update_icon(UPDATE_OVERLAYS)
+
+/datum/element/rust/Detach(atom/source, ...)
+	. = ..()
+	UnregisterSignal(source,\
+		list(COMSIG_ATOM_UPDATE_OVERLAYS, COMSIG_PARENT_EXAMINE, COMSIG_ATOM_SECONDARY_TOOL_ACT(TOOL_WELDER), COMSIG_ATOM_SECONDARY_TOOL_ACT(TOOL_RUSTSCRAPER)))
```

Comment:
Split this up into new lines, easier to read and append to.

Reply:
Sir yes sir oorah